### PR TITLE
[Marked] Add inlineLexer function to definitions

### PR DESCRIPTION
**Pull request by mwickett**
Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

We use the `inlineLexer` method and I noticed it wasn't included in the definitions, so this adds it. I think it's fairly straightforward, but please let me know if I'm missing anything.

Unfortunately the `marked` docs aren't comprehensive and so this method isn't documented anywhere.

### CODE REVIEW

**Comment by worr** [comment]
Hey there, can you link the definition from marked? I see an`InlineLexer` object, but that would need to be stubbed differently.

**Comment by mwickett** [comment]
Sorry, should have included that. `marked.inlineLexer` is a reference to `output` from `InlineLexer` - see https://github.com/markedjs/marked/blob/939b637a1631cc5f1f7790e4b4bdf64ca281dfb2/lib/marked.js#L666

File: types/marked/index.d.ts [hunk 1]
```diff
@@ -34,6 +35,15 @@ declare namespace marked {
...
     function lexer(src: string, options?: MarkedOptions): TokensList;
 
+    /**
+     * @param src String of markdown source to be compiled
+     * @param links Array of links
+     * @param options Hash of options
+     * @return String of compiled HTML
+     */
+
+    function inlineLexer(src: string, links?: string[], options?: MarkedOptions): string;
```
Comment (worr) [inline types/marked/index.d.ts]: It looks like `links` should be required here

**Review by worr — CHANGES_REQUESTED** [review]

File: types/marked/index.d.ts (hunk 1 above)
Comment (mwickett) [reply types/marked/index.d.ts]: Whoops, absolutely right. I'll update.

Comment (mwickett) [reply types/marked/index.d.ts]: Should be good to go now.

**Review by worr — APPROVED** [review]

**Comment by worr** [comment]
Looks good. Thanks a ton!!

**Pull request merged**